cmd/controller-manager/app/controller-manager.go

```diff
@@ -117,6 +117,17 @@ func Run(opts *options.Options) error {
 		klog.Fatalf("Invalid Feature Gate: %v", err)
 	}
 
+	// for in-cluster deployment set the namespace associated
```

Comment:
Discovery of the kubefed namespace should only be attempted if it was not supplied explicitly and the configuration indicates that the controller manager is running inside a cluster i.e. the following conditions:

 - `len(opts.Config.FederationNamespace) == 0`
 - `len(masterURL) == 0`
 - `len(kubeconfig) == 0`

Reply:
Thanks, @marun. I will update this PR to check for this condition before setting up the value for the `federation namespace`.